File: 119-pascals-triangle-ii/119-pascals-triangle-ii.cpp
class Solution {
public:
    
    int dp[35][35];
    int get(int r,int c)
    {
        if(r<0 || c<0 || c>r)
            return 0;
        
        if(r==0 && c==0)
            return 1;
        
        if(dp[r][c])
            return dp[r][c];
        return  dp[r][c]=get(r-1,c-1)+get(r-1,c);
    }
    vector<int> getRow(int rowIndex) {
        vector<int> ans;
        
        for(int i=0;i<rowIndex+1;i++)
        {
            int x=get(rowIndex,i);
            ans.push_back(x);
        }
        
        
        return ans;
    }
};

File: 119-pascals-triangle-ii/119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        pasc = [1]*(rowIndex+1)
        for i in range(2 ,rowIndex+1):
            for j in range(i-1, 0, -1):
                pasc[j] += pasc[j-1]
        return pasc
        